src/Plugin/Sampler/Bundle.php

```diff
@@ -133,94 +144,24 @@ public function collect() {
 
       /** @var \Drupal\Core\Field\FieldConfigInterface $fieldConfig */
       foreach ($fields as $fieldConfig) {
+        $fieldType = $fieldConfig->getType();
+        if (!isset($this->collectedData[$mapping]['fields'][$fieldType])) {
+          $this->collectedData[$mapping]['fields'][$fieldType] = [];
+        }
+
+        $fieldData = $this->fieldData->defaultFieldData($fieldConfig);
```

Comment:
Just one idea.

Can we introduce function on `FieldData` that will return filed data? Then we can pull all custom logic about fields inside. For example: `if (in_array($fieldConfig->getType(), ['entity_reference', 'entity_reference_revisions'])) {`

That would make the interface to `FieldData` service simple and all specific logic to define what data should be used, etc. would be inside service.

At end it would be just: `$this->collectedData[$mapping]['fields'][$fieldType][] = $this->fieldData->getFieldData($fieldConfig);`

And we can keep other function protected.